src/resources/views/crud/inc/form_fields_script.blade.php

```diff
@@ -66,9 +66,12 @@ class CrudField {
                 return this;
             }
 
-            this.input?.addEventListener('input', fieldChanged, false);
-            this.$input.change(fieldChanged);
-
+            if(typeof this.input !== 'undefined') {
+                if(['text', 'textarea'].includes(this.input.getAttribute('type'))) {
+                    this.input?.addEventListener('input', fieldChanged, false);
+                }
+                this.$input.change(fieldChanged);
+            }
```

Comment:
Wait... can `this.input` even be undefined? Is that possible? And if it is, shouldn't we throw a custom error explaining that?

Would something like this also work @pxpm ?

```suggestion
            if (typeof this.input == 'undefined') {
                console.log('CrudField Error: Could not find main input for field '.this.name);
            }

            // trigger field change on input change
            this.$input.change(fieldChanged);

            // trigger field change when typing inside an input or textarea
            if(['text', 'textarea'].includes(this.input.getAttribute('type'))) {
                this.input?.addEventListener('input', fieldChanged, false);
            }
```

Reply:
The error is already thrown, but it does not stop the execution. 

That's what `?` is doing there after `this.input` . It can be removed if we merge this.

Reply:
Un-nesting the if condition does not raise an error, but technically is not correct to "try to add an event" to an inexistent input. jQuery does not complain ... but still.